### PR TITLE
Pear - Add prompt parameter on pear module that allows to specify one or more prompts

### PR DESCRIPTION
**Pull request by tchernomax**
# SUMMARY

This is a copy of https://github.com/ansible-collections/community.general/pull/233

Add a new parameter to pear module allowing user to pass one or more regex and (optionnally) their answer for each package that will be installed.
Depending on what package we want to install, pear can ask for some information or just typing a character.
This hangs Ansible execution and the added parameter is used to detect these questions and answer them with a newline character by default or a user specified string.

# ISSUE TYPE

* Bugfix Pull Request

# COMPONENT NAME

pear

# ADDITIONAL INFORMATION

This fixes an issue created on repo 2 and half years ago :

Exemple playbook:

The example below have a prompt specified but no answer. Thus, the module will input newline caracter "\n" by default

```yaml
pear:
  name: pecl/apcu
  state: installed
  prompts:
    - (.*)Enable internal debugging in APCu \[no\]
```

The example below is the same except we specify a user answer.

```yaml
pear:
  name: pecl/apcu
  state: installed
  prompts:
    - (.*)Enable internal debugging in APCu \[no\]: "yes"
```

### CODE REVIEW

File: changelogs/fragments/29253-pear_add_prompts_parameter.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- pear - added ``prompts`` parameter to allow users to specify expected prompt that could hang Ansible execution
```
Comment (felixfontein) [inline changelogs/fragments/29253-pear_add_prompts_parameter.yml:2]:
```suggestion
- pear - added ``prompts`` parameter to allow users to specify expected prompt that could hang Ansible execution (https://github.com/ansible-collections/community.general/pull/530).
```

File: plugins/modules/packaging/language/pear.py [hunk 2]
```diff
@@ -31,11 +31,24 @@
...
+            - Prompts will be processed in the same order as the packages list
+            - You can optionnally specify an answer to any question in the list
+            - If no answer is provided, the list item will only contain the regular expression.
+            - "To specify an answer, the item will be a dict with the regular expression as key and the answer as value C(my_regular_expression: 'an_answer')"
+            - You can provide a list containing items with or without answer
+            - A prompt list can be shorter or longer than the packages list but will issue a warning
+            - If you want to specify that a package will not need prompts in the middle of a list,  C(null)
+        type: list
+        elements: raw
+        version_added: "2.10"
```
Comment (felixfontein) [inline plugins/modules/packaging/language/pear.py:48]:
```suggestion
        version_added: 0.2.0
```

File: plugins/modules/packaging/language/pear.py [hunk 3]
```diff
@@ -31,11 +31,24 @@
...
         choices: ["present", "absent", "latest"]
     executable:
-      description:
-        - Path to the pear executable
+        description:
+            - Path to the pear executable
+    prompts:
+        description:
+            - List of regular expressions that can be used to detect prompts during pear package installation to answer the expected question.
+            - Prompts will be processed in the same order as the packages list
```
Comment (felixfontein) [inline plugins/modules/packaging/language/pear.py:39]:
```suggestion
            - Prompts will be processed in the same order as the packages list.
```
(Same for next ones.)

File: plugins/modules/packaging/language/pear.py [hunk 4]
```diff
@@ -31,11 +31,24 @@
         default: "present"
         choices: ["present", "absent", "latest"]
     executable:
-      description:
-        - Path to the pear executable
+        description:
+            - Path to the pear executable
```
Comment (felixfontein) [inline plugins/modules/packaging/language/pear.py:35]:
```suggestion
            - Path to the pear executable.
```

**Comment by tchernomax** [comment]
@felixfontein thanks for the review. patch applied.

**Comment by Veltarn** [comment]
Thanks @tchernomax

**Comment by tchernomax** [comment]
> @tchernomax can this be merged, or are you waiting for more reviews?

@felixfontein I think it can be merged.
I already use it in some of my role and everything works fine.

**Comment by felixfontein** [comment]
So may it be then ;) Thanks @Veltarn and @tchernomax!

**Pull request merged**